frontend/components/tree/js/tree.service.ts

```diff
@@ -1049,14 +1049,9 @@ export class TreeService {
 
 		for (const id in this.currentSelectedNodes) {
 
-			if (!id) { continue; }
-
 			const currentNode = this.currentSelectedNodes[id];
 
-			// Skip over any node that has already been visited
-			if (visitedNodes[currentNode._id]) {
-				continue;
-			}
+			if (!id || visitedNodes[currentNode._id]) { continue; }
```

Comment:
Actually, move this up before L1052 and do:
````js
if (!id || visitedNodes[id]) { continue; }
````

provided currentNode._id is the same as id (which sounds like it should be?)